### DISCUSSION TITLE
Import 3D inpainted OBJ files into Unreal ?

### DISCUSSION THREAD

**Discussion by adamcolestudio** [thygate/stable-diffusion-webui-depthmap-script#173] [Announcements]
Hello! When I try to import a 3D inpainted OBJ file into Unreal engine, I get a mess or vertices and I can't quite work out how to view it correctly.

Could you potentially add instructions or a video for how best to use these assets in Unreal? Unsure if I'm misusing the obj file or there is some other issue.

<img width="810" alt="Screen Shot 2023-04-12 at 7 16 18 PM" src="https://user-images.githubusercontent.com/5685294/231549313-b83a5aac-b91d-46b3-a8e8-91504f0ef601.png">

(ps. love the repo, thanks so much for your effors 🙌)

**Comment by thygate** [comment]
See README section on blender, same applies here.  
https://github.com/thygate/stable-diffusion-webui-depthmap-script#Blender

Also, this is a mesh with vertex colors, not a textured mesh, unless you're going to use a shader that supports vertex color you will have to create a textured mesh yourself.

The simple way to get SD images with depth into any 3D engine is the old plane with displacement modifier depthmap and rgb texture map ..

**Comment by adamcolestudio** [comment]
Thanks so much for the quick update and general support. 

I've been able to get the camera aligned but can't figure out the correct combination of lighting and materials to get anything resembling the original image or example videos. I am able to see the shape and depth of the image through the camera though.

In both Blender and Unreal I've set up the mesh to use vertex colors but had no luck. If anyone has figured out a workflow for this, please do share it. Thanks 🙏 


Original image:
![HD landscape 01](https://user-images.githubusercontent.com/5685294/233156793-2d43400f-10aa-4619-803e-b2c4c1053728.png)

Blender attempt:
<img width="987" alt="Screen Shot 2023-04-19 at 5 53 33 PM" src="https://user-images.githubusercontent.com/5685294/233156871-5e51ba97-f073-4171-ba5f-8fa92fb2fc04.png">

Unreal Engine attempt:
![IMG_4167](https://user-images.githubusercontent.com/5685294/233158693-d8ef6da6-404f-46b4-a1bf-84254e1bfbbc.JPG)

**Comment by thygate** [comment]
Try the emission shader in blender..

![image](https://user-images.githubusercontent.com/96586282/233173181-7bd71cb2-5937-476e-9275-faa1d8ffb14c.png)

**Reply by thygate** [reply]
I have not used Unreal engine, but keep in mind the Wavefront OBJ file format the extension exports is not standard, as the original standard did not include a way to include vertex colors. Some packages like blender DO support it, but Unreal might not, try the PLY file format there instead. In Settings, enable the additional ply option.

**Reply by thygate** [reply]
confirmed, Unreal does not support vertex color in Wavefront OBJ, so use the PLY format, or convert the OBJ to something like FBX in blender ..

**Comment by thygate** [comment]
This type of picture isn't going to give any special results from 3D in-painting, since there is nothing in the foreground, and the mountains are so far away you can hardly perceive depth at that range ..

The result is going to be the same as just displacing a plane with the depthmap.
